cpp/tests/reductions/ewm_tests.cpp

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "scan_tests.hpp"
+
+#include <cudf_test/column_utilities.hpp>
+#include <cudf_test/column_wrapper.hpp>
+#include <cudf_test/iterator_utilities.hpp>
+
+#include <cudf/detail/aggregation/aggregation.hpp>
+#include <cudf/reduction.hpp>
+
+using aggregation = cudf::aggregation;
+using cudf::null_policy;
+using cudf::scan_type;
+using namespace cudf::test::iterators;
+
+template <typename T>
+struct TypedEwmScanTest : BaseScanTest<T> {
+  inline void test_ungrouped_ewma_scan(cudf::column_view const& input,
+                                       cudf::column_view const& expect_vals,
+                                       std::unique_ptr<aggregation> const& agg,
+                                       null_policy null_handling)
+  {
+    auto col_out = cudf::scan(input, agg, scan_type::INCLUSIVE, null_handling);
+    CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(expect_vals, col_out->view());
+  }
+};
+
+TYPED_TEST_SUITE(TypedEwmScanTest, cudf::test::FloatingPointTypes);
+
```

Comment:
I am in support of larger tests, however coding up the serial version of the algorithm isn't trivial, especially since we have four different versions of it in this PR. The problem will worsen when `ewmvar` and `ewmstd` are introduced, since those act differently depending on an additional parameter, `bias`. 

In lieu of such would you be ok with perhaps loading some static data generated from pandas? What do you think @harrism ?

Reply:
@harrism what do you think here? IMO this issue is similar to what cuIO faces where in practice it's just much easier to test larger and more complex data from Python, including the cases where we need to test multiple thread blocks etc.

Reply:
OK.